<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Print usage of hisat2 — hisat2_usage • Rhisat2</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Print usage of hisat2 — hisat2_usage"><meta property="og:description" content="Print usage of hisat2"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rhisat2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/Rhisat2.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/fmicompbio/Rhisat2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Print usage of hisat2</h1>
    <small class="dont-index">Source: <a href="https://github.com/fmicompbio/Rhisat2/blob/HEAD/R/Rhisat2.R" class="external-link"><code>R/Rhisat2.R</code></a></small>
    <div class="hidden name"><code>hisat2_usage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Print usage of hisat2</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">hisat2_usage</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="value">
    <h2>Value</h2>
    

<p>No value is returned, the usage of <code>hisat2</code> is printed to the
  console.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Charlotte Soneson</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">hisat2_usage</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] "HISAT2 version 2.2.1 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)"                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2] "Usage: "                                                                                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3] "  hisat2-align [options]* -x &lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]"                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5] "  &lt;ht2-idx&gt;  Index filename prefix (minus trailing .X.ht2)."                                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6] "  &lt;m1&gt;       Files with #1 mates, paired with files in &lt;m2&gt;."                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [7] "  &lt;m2&gt;       Files with #2 mates, paired with files in &lt;m1&gt;."                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [8] "  &lt;r&gt;        Files with unpaired reads."                                                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9] "  &lt;sam&gt;      File for SAM output (default: stdout)"                                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [10] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [11] "  &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and can be"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [12] "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'."                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [13] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [14] "Options (defaults in parentheses):"                                                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [15] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [16] " Input:"                                                                                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [17] "  -q                 query input files are FASTQ .fq/.fastq (default)"                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [18] "  --qseq             query input files are in Illumina's qseq format"                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [19] "  -f                 query input files are (multi-)FASTA .fa/.mfa"                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [20] "  -r                 query input files are raw one-sequence-per-line"                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [21] "  -c                 &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files"                                                                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [22] "  -s/--skip &lt;int&gt;    skip the first &lt;int&gt; reads/pairs in the input (none)"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [23] "  -u/--upto &lt;int&gt;    stop after first &lt;int&gt; reads/pairs (no limit)"                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [24] "  -5/--trim5 &lt;int&gt;   trim &lt;int&gt; bases from 5'/left end of reads (0)"                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [25] "  -3/--trim3 &lt;int&gt;   trim &lt;int&gt; bases from 3'/right end of reads (0)"                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [26] "  --phred33          qualities are Phred+33 (default)"                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [27] "  --phred64          qualities are Phred+64"                                                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [28] "  --int-quals        qualities encoded as space-delimited integers"                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [29] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [30] " Presets:                 Same as:"                                                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [31] "   --fast                 --no-repeat-index"                                                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [32] "   --sensitive            --bowtie2-dp 1 -k 30 --score-min L,0,-0.5"                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [33] "   --very-sensitive       --bowtie2-dp 2 -k 50 --score-min L,0,-1"                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [34] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [35] " Alignment:"                                                                                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [36] "  --bowtie2-dp &lt;int&gt; use Bowtie2's dynamic programming alignment algorithm (0) - 0: no dynamic programming, 1: conditional dynamic programming, and 2: unconditional dynamic programming (slowest)"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [37] "  --n-ceil &lt;func&gt;    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)"                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [38] "  --ignore-quals     treat all quality values as 30 on Phred scale (off)"                                                                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [39] "  --nofw             do not align forward (original) version of read (off)"                                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [40] "  --norc             do not align reverse-complement version of read (off)"                                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [41] "  --no-repeat-index  do not use repeat index"                                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [42] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [43] " Spliced Alignment:"                                                                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [44] "  --pen-cansplice &lt;int&gt;              penalty for a canonical splice site (0)"                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [45] "  --pen-noncansplice &lt;int&gt;           penalty for a non-canonical splice site (12)"                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46] "  --pen-canintronlen &lt;func&gt;          penalty for long introns (G,-8,1) with canonical splice sites"                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [47] "  --pen-noncanintronlen &lt;func&gt;       penalty for long introns (G,-8,1) with noncanonical splice sites"                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [48] "  --min-intronlen &lt;int&gt;              minimum intron length (20)"                                                                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [49] "  --max-intronlen &lt;int&gt;              maximum intron length (500000)"                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [50] "  --known-splicesite-infile &lt;path&gt;   provide a list of known splice sites"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [51] "  --novel-splicesite-outfile &lt;path&gt;  report a list of splice sites"                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [52] "  --novel-splicesite-infile &lt;path&gt;   provide a list of novel splice sites"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [53] "  --no-temp-splicesite               disable the use of splice sites found"                                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [54] "  --no-spliced-alignment             disable spliced alignment"                                                                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [55] "  --rna-strandness &lt;string&gt;          specify strand-specific information (unstranded)"                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [56] "  --tmo                              reports only those alignments within known transcriptome"                                                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [57] "  --dta                              reports alignments tailored for transcript assemblers"                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [58] "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks"                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [59] "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)"                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [60] "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads"                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [61] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [62] " Scoring:"                                                                                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [63] "  --mp &lt;int&gt;,&lt;int&gt;   max and min penalties for mismatch; lower qual = lower penalty &lt;6,2&gt;"                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [64] "  --sp &lt;int&gt;,&lt;int&gt;   max and min penalties for soft-clipping; lower qual = lower penalty &lt;2,1&gt;"                                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [65] "  --no-softclip      no soft-clipping"                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [66] "  --np &lt;int&gt;         penalty for non-A/C/G/Ts in read/ref (1)"                                                                                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [67] "  --rdg &lt;int&gt;,&lt;int&gt;  read gap open, extend penalties (5,3)"                                                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [68] "  --rfg &lt;int&gt;,&lt;int&gt;  reference gap open, extend penalties (5,3)"                                                                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [69] "  --score-min &lt;func&gt; min acceptable alignment score w/r/t read length"                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [70] "                     (L,0.0,-0.2)"                                                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [71] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [72] " Reporting:"                                                                                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [73] "  -k &lt;int&gt;           It searches for at most &lt;int&gt; distinct, primary alignments for each read. Primary alignments mean "                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [74] "                     alignments whose alignment score is equal to or higher than any other alignments. The search terminates "                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [75] "                     when it cannot find more distinct valid alignments, or when it finds &lt;int&gt;, whichever happens first. "                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [76] "                     The alignment score for a paired-end alignment equals the sum of the alignment scores of "                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [77] "                     the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit "                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [78] "                     (which equals 256) set in its FLAGS field. For reads that have more than &lt;int&gt; distinct, "                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [79] "                     valid alignments, hisat2 does not guarantee that the &lt;int&gt; alignments reported are the best possible "                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [80] "                     in terms of alignment score. Default: 5 (linear index) or 10 (graph index)."                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [81] "                     Note: HISAT2 is not designed with large values for -k in mind, and when aligning reads to long, "                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [82] "                     repetitive genomes, large -k could make alignment much slower."                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [83] "  --max-seeds &lt;int&gt;  HISAT2, like other aligners, uses seed-and-extend approaches. HISAT2 tries to extend seeds to "                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [84] "                     full-length alignments. In HISAT2, --max-seeds is used to control the maximum number of seeds that "                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [85] "                     will be extended. For DNA-read alignment (--no-spliced-alignment), HISAT2 extends up to these many seeds"                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [86] "                     and skips the rest of the seeds. For RNA-read alignment, HISAT2 skips extending seeds and reports "                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [87] "                     no alignments if the number of seeds is larger than the number specified with the option, "                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [88] "                     to be compatible with previous versions of HISAT2. Large values for --max-seeds may improve alignment "                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [89] "                     sensitivity, but HISAT2 is not designed with large values for --max-seeds in mind, and when aligning "                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [90] "                     reads to long, repetitive genomes, large --max-seeds could make alignment much slower. "                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91] "                     The default value is the maximum of 5 and the value that comes with -k times 2."                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [92] "  -a/--all           HISAT2 reports all alignments it can find. Using the option is equivalent to using both --max-seeds "                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [93] "                     and -k with the maximum value that a 64-bit signed integer can represent (9,223,372,036,854,775,807)."                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [94] "  --repeat           report alignments to repeat sequences directly"                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [95] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [96] " Paired-end:"                                                                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [97] "  -I/--minins &lt;int&gt;  minimum fragment length (0), only valid with --no-spliced-alignment"                                                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [98] "  -X/--maxins &lt;int&gt;  maximum fragment length (500), only valid with --no-spliced-alignment"                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [99] "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)"                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [100] "  --no-mixed         suppress unpaired alignments for paired reads"                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [101] "  --no-discordant    suppress discordant alignments for paired reads"                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [102] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [103] " Output:"                                                                                                                                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [104] "  -t/--time          print wall-clock time taken by search phases"                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [105] "  --summary-file &lt;path&gt; print alignment summary to this file."                                                                                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [106] "  --new-summary         print alignment summary in a new style, which is more machine-friendly."                                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [107] "  --quiet               print nothing to stderr except serious errors"                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [108] "  --met-file &lt;path&gt;     send metrics to file at &lt;path&gt; (off)"                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [109] "  --met-stderr          send metrics to stderr (off)"                                                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [110] "  --met &lt;int&gt;           report internal counters &amp; metrics every &lt;int&gt; secs (1)"                                                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [111] "  --no-head             suppress header lines, i.e. lines starting with @"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [112] "  --no-sq               suppress @SQ header lines"                                                                                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [113] "  --rg-id &lt;text&gt;        set read group id, reflected in @RG line and RG:Z: opt field"                                                                                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [114] "  --rg &lt;text&gt;           add &lt;text&gt; (\"lab:value\") to @RG line of SAM header."                                                                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [115] "                        Note: @RG line only printed when --rg-id is set."                                                                                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [116] "  --omit-sec-seq        put '*' in SEQ and QUAL fields for secondary alignments."                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [117] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [118] " Performance:"                                                                                                                                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [119] "  -o/--offrate &lt;int&gt; override offrate of index; must be &gt;= index's offrate"                                                                                                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [120] "  -p/--threads &lt;int&gt; number of alignment threads to launch (1)"                                                                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [121] "  --reorder          force SAM output order to match order of input reads"                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [122] "  --mm               use memory-mapped I/O for index; many 'hisat2's can share"                                                                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [123] ""                                                                                                                                                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [124] " Other:"                                                                                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [125] "  --qc-filter        filter out reads that are bad according to QSEQ filter"                                                                                                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [126] "  --seed &lt;int&gt;       seed for random number generator (0)"                                                                                                                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [127] "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes"                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [128] "  --remove-chrname   remove 'chr' from reference names in alignment"                                                                                                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [129] "  --add-chrname      add 'chr' to reference names in alignment "                                                                                                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [130] "  --version          print version information and quit"                                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [131] "  -h/--help          print this usage message"                                                                                                                                                     </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Charlotte Soneson.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

