<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Print usage of hisat2 — hisat2_usage • Rhisat2</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Print usage of hisat2 — hisat2_usage" />
<meta property="og:description" content="Print usage of hisat2" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rhisat2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/Rhisat2.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fmicompbio/Rhisat2/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Print usage of hisat2</h1>
    <small class="dont-index">Source: <a href='https://github.com/fmicompbio/Rhisat2/blob/master/R/Rhisat2.R'><code>R/Rhisat2.R</code></a></small>
    <div class="hidden name"><code>hisat2_usage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Print usage of hisat2</p>
    </div>

    <pre class="usage"><span class='fu'>hisat2_usage</span><span class='op'>(</span><span class='op'>)</span></pre>


    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value is returned, the usage of <code>hisat2</code> is printed to the
  console.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Charlotte Soneson</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>hisat2_usage</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   [1] "HISAT2 version 2.1.0 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)"                  
#&gt;   [2] "Usage: "                                                                                                    
#&gt;   [3] "  hisat2-align [options]* -x &lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]"                               
#&gt;   [4] ""                                                                                                           
#&gt;   [5] "  &lt;ht2-idx&gt;  Index filename prefix (minus trailing .X.ht2)."                                                
#&gt;   [6] "  &lt;m1&gt;       Files with #1 mates, paired with files in &lt;m2&gt;."                                               
#&gt;   [7] "  &lt;m2&gt;       Files with #2 mates, paired with files in &lt;m1&gt;."                                               
#&gt;   [8] "  &lt;r&gt;        Files with unpaired reads."                                                                    
#&gt;   [9] "  &lt;sam&gt;      File for SAM output (default: stdout)"                                                         
#&gt;  [10] ""                                                                                                           
#&gt;  [11] "  &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and can be"                                  
#&gt;  [12] "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'."                                          
#&gt;  [13] ""                                                                                                           
#&gt;  [14] "Options (defaults in parentheses):"                                                                         
#&gt;  [15] ""                                                                                                           
#&gt;  [16] " Input:"                                                                                                    
#&gt;  [17] "  -q                 query input files are FASTQ .fq/.fastq (default)"                                      
#&gt;  [18] "  --qseq             query input files are in Illumina's qseq format"                                       
#&gt;  [19] "  -f                 query input files are (multi-)FASTA .fa/.mfa"                                          
#&gt;  [20] "  -r                 query input files are raw one-sequence-per-line"                                       
#&gt;  [21] "  -c                 &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files"                                   
#&gt;  [22] "  -s/--skip &lt;int&gt;    skip the first &lt;int&gt; reads/pairs in the input (none)"                                  
#&gt;  [23] "  -u/--upto &lt;int&gt;    stop after first &lt;int&gt; reads/pairs (no limit)"                                         
#&gt;  [24] "  -5/--trim5 &lt;int&gt;   trim &lt;int&gt; bases from 5'/left end of reads (0)"                                        
#&gt;  [25] "  -3/--trim3 &lt;int&gt;   trim &lt;int&gt; bases from 3'/right end of reads (0)"                                       
#&gt;  [26] "  --phred33          qualities are Phred+33 (default)"                                                      
#&gt;  [27] "  --phred64          qualities are Phred+64"                                                                
#&gt;  [28] "  --int-quals        qualities encoded as space-delimited integers"                                         
#&gt;  [29] ""                                                                                                           
#&gt;  [30] " Alignment:"                                                                                                
#&gt;  [31] "  --n-ceil &lt;func&gt;    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)"                               
#&gt;  [32] "  --ignore-quals     treat all quality values as 30 on Phred scale (off)"                                   
#&gt;  [33] "  --nofw             do not align forward (original) version of read (off)"                                 
#&gt;  [34] "  --norc             do not align reverse-complement version of read (off)"                                 
#&gt;  [35] ""                                                                                                           
#&gt;  [36] " Spliced Alignment:"                                                                                        
#&gt;  [37] "  --pen-cansplice &lt;int&gt;              penalty for a canonical splice site (0)"                               
#&gt;  [38] "  --pen-noncansplice &lt;int&gt;           penalty for a non-canonical splice site (12)"                          
#&gt;  [39] "  --pen-canintronlen &lt;func&gt;          penalty for long introns (G,-8,1) with canonical splice sites"         
#&gt;  [40] "  --pen-noncanintronlen &lt;func&gt;       penalty for long introns (G,-8,1) with noncanonical splice sites"      
#&gt;  [41] "  --min-intronlen &lt;int&gt;              minimum intron length (20)"                                            
#&gt;  [42] "  --max-intronlen &lt;int&gt;              maximum intron length (500000)"                                        
#&gt;  [43] "  --known-splicesite-infile &lt;path&gt;   provide a list of known splice sites"                                  
#&gt;  [44] "  --novel-splicesite-outfile &lt;path&gt;  report a list of splice sites"                                         
#&gt;  [45] "  --novel-splicesite-infile &lt;path&gt;   provide a list of novel splice sites"                                  
#&gt;  [46] "  --no-temp-splicesite               disable the use of splice sites found"                                 
#&gt;  [47] "  --no-spliced-alignment             disable spliced alignment"                                             
#&gt;  [48] "  --rna-strandness &lt;string&gt;          specify strand-specific information (unstranded)"                      
#&gt;  [49] "  --tmo                              reports only those alignments within known transcriptome"              
#&gt;  [50] "  --dta                              reports alignments tailored for transcript assemblers"                 
#&gt;  [51] "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks"                
#&gt;  [52] "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)\023"
#&gt;  [53] "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads"                 
#&gt;  [54] ""                                                                                                           
#&gt;  [55] " Scoring:"                                                                                                  
#&gt;  [56] "  --mp &lt;int&gt;,&lt;int&gt;   max and min penalties for mismatch; lower qual = lower penalty &lt;6,2&gt;"                  
#&gt;  [57] "  --sp &lt;int&gt;,&lt;int&gt;   max and min penalties for soft-clipping; lower qual = lower penalty &lt;2,1&gt;"             
#&gt;  [58] "  --no-softclip      no soft-clipping"                                                                      
#&gt;  [59] "  --np &lt;int&gt;         penalty for non-A/C/G/Ts in read/ref (1)"                                              
#&gt;  [60] "  --rdg &lt;int&gt;,&lt;int&gt;  read gap open, extend penalties (5,3)"                                                 
#&gt;  [61] "  --rfg &lt;int&gt;,&lt;int&gt;  reference gap open, extend penalties (5,3)"                                            
#&gt;  [62] "  --score-min &lt;func&gt; min acceptable alignment score w/r/t read length"                                      
#&gt;  [63] "                     (L,0.0,-0.2)"                                                                          
#&gt;  [64] ""                                                                                                           
#&gt;  [65] " Reporting:"                                                                                                
#&gt;  [66] "  -k &lt;int&gt; (default: 5) report up to &lt;int&gt; alns per read"                                                   
#&gt;  [67] ""                                                                                                           
#&gt;  [68] " Paired-end:"                                                                                               
#&gt;  [69] "  -I/--minins &lt;int&gt;  minimum fragment length (0), only valid with --no-spliced-alignment"                   
#&gt;  [70] "  -X/--maxins &lt;int&gt;  maximum fragment length (500), only valid with --no-spliced-alignment"                 
#&gt;  [71] "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)"                                       
#&gt;  [72] "  --no-mixed         suppress unpaired alignments for paired reads"                                         
#&gt;  [73] "  --no-discordant    suppress discordant alignments for paired reads"                                       
#&gt;  [74] ""                                                                                                           
#&gt;  [75] " Output:"                                                                                                   
#&gt;  [76] "  -t/--time          print wall-clock time taken by search phases"                                          
#&gt;  [77] "  --summary-file     print alignment summary to this file."                                                 
#&gt;  [78] "  --new-summary      print alignment summary in a new style, which is more machine-friendly."               
#&gt;  [79] "  --quiet            print nothing to stderr except serious errors"                                         
#&gt;  [80] "  --met-file &lt;path&gt;  send metrics to file at &lt;path&gt; (off)"                                                  
#&gt;  [81] "  --met-stderr       send metrics to stderr (off)"                                                          
#&gt;  [82] "  --met &lt;int&gt;        report internal counters &amp; metrics every &lt;int&gt; secs (1)"                               
#&gt;  [83] "  --no-head          supppress header lines, i.e. lines starting with @"                                    
#&gt;  [84] "  --no-sq            supppress @SQ header lines"                                                            
#&gt;  [85] "  --rg-id &lt;text&gt;     set read group id, reflected in @RG line and RG:Z: opt field"                          
#&gt;  [86] "  --rg &lt;text&gt;        add &lt;text&gt; (\"lab:value\") to @RG line of SAM header."                                 
#&gt;  [87] "                     Note: @RG line only printed when --rg-id is set."                                      
#&gt;  [88] "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments."                              
#&gt;  [89] ""                                                                                                           
#&gt;  [90] " Performance:"                                                                                              
#&gt;  [91] "  -o/--offrate &lt;int&gt; override offrate of index; must be &gt;= index's offrate"                                 
#&gt;  [92] "  -p/--threads &lt;int&gt; number of alignment threads to launch (1)"                                             
#&gt;  [93] "  --reorder          force SAM output order to match order of input reads"                                  
#&gt;  [94] "  --mm               use memory-mapped I/O for index; many 'hisat2's can share"                             
#&gt;  [95] ""                                                                                                           
#&gt;  [96] " Other:"                                                                                                    
#&gt;  [97] "  --qc-filter        filter out reads that are bad according to QSEQ filter"                                
#&gt;  [98] "  --seed &lt;int&gt;       seed for random number generator (0)"                                                  
#&gt;  [99] "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes"                         
#&gt; [100] "  --remove-chrname   remove 'chr' from reference names in alignment"                                        
#&gt; [101] "  --add-chrname      add 'chr' to reference names in alignment "                                            
#&gt; [102] "  --version          print version information and quit"                                                    
#&gt; [103] "  -h/--help          print this usage message"                                                              </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Charlotte Soneson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


