<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Aligning reads with Rhisat2 • Rhisat2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Aligning reads with Rhisat2">
<meta property="og:description" content="Rhisat2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rhisat2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Rhisat2.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fmicompbio/Rhisat2/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Aligning reads with Rhisat2</h1>
                        <h4 data-toc-skip class="author">Charlotte Soneson</h4>
            
            <h4 data-toc-skip class="date">2022-10-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fmicompbio/Rhisat2/blob/HEAD/vignettes/Rhisat2.Rmd" class="external-link"><code>vignettes/Rhisat2.Rmd</code></a></small>
      <div class="hidden name"><code>Rhisat2.Rmd</code></div>

    </div>

    
    
<p>The <em><a href="https://bioconductor.org/packages/3.16/Rhisat2" class="external-link">Rhisat2</a></em> package provides an interface to the <a href="https://ccb.jhu.edu/software/hisat2/index.shtml" class="external-link">HISAT2</a> short-read aligner software <span class="citation">(Kim, Langmead, and Salzberg 2015)</span>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Use the <code>BiocManager</code> package to download and install the package from Bioconductor as follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"Rhisat2"</span><span class="op">)</span></span></code></pre></div>
<p>If required, the latest development version of the package can also be installed from GitHub.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"fmicompbio/Rhisat2"</span><span class="op">)</span></span></code></pre></div>
<p>Once the package is installed, load it into your R session:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fmicompbio/Rhisat2" class="external-link">Rhisat2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="building-a-genome-index">Building a genome index<a class="anchor" aria-label="anchor" href="#building-a-genome-index"></a>
</h2>
<p>To build an index for the alignment, use the <code>hisat2_build</code> function. You need to provide one or more fasta file with reference sequences, as well as an output directory where the index will be stored, and a “prefix” (that will determine the name of the index files in the output directory). Any additional arguments to the <code>hisat2-build</code> binary can also be supplied to the function (see <code><a href="../reference/hisat2_build_usage.html">hisat2_build_usage()</a></code> or <a href="https://ccb.jhu.edu/software/hisat2/manual.shtml#the-hisat2-build-indexer" class="external-link">https://ccb.jhu.edu/software/hisat2/manual.shtml#the-hisat2-build-indexer</a> for an overview of the available options).</p>
<p>The package contains three example reference sequences, corresponding to short pieces of three different chromosomes. We show how to create an index (named <code>myindex</code>) based on these sequences.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/refs"</span>, package<span class="op">=</span><span class="st">"Rhisat2"</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">"\\.fa$"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "chr1.fa" "chr2.fa" "chr3.fa"</span></span>
<span><span class="va">refs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/refs"</span>, package<span class="op">=</span><span class="st">"Rhisat2"</span><span class="op">)</span>, </span>
<span>                   full.names<span class="op">=</span><span class="cn">TRUE</span>, pattern<span class="op">=</span><span class="st">"\\.fa$"</span><span class="op">)</span></span>
<span><span class="va">td</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/hisat2_build.html">hisat2_build</a></span><span class="op">(</span>references<span class="op">=</span><span class="va">refs</span>, outdir<span class="op">=</span><span class="va">td</span>, prefix<span class="op">=</span><span class="st">"myindex"</span>, </span>
<span>             force<span class="op">=</span><span class="cn">TRUE</span>, strict<span class="op">=</span><span class="cn">TRUE</span>, execute<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>By instead setting <code>execute=FALSE</code> in the command above, <code><a href="../reference/hisat2_build.html">hisat2_build()</a></code> will return the index building shell command for inspection, without executing it.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="../reference/hisat2_build.html">hisat2_build</a></span><span class="op">(</span>references<span class="op">=</span><span class="va">refs</span>, outdir<span class="op">=</span><span class="va">td</span>, prefix<span class="op">=</span><span class="st">"myindex"</span>,</span>
<span>                   force<span class="op">=</span><span class="cn">TRUE</span>, strict<span class="op">=</span><span class="cn">TRUE</span>, execute<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "'/Users/runner/work/_temp/Library/Rhisat2/hisat2-build' '/Users/runner/work/_temp/Library/Rhisat2/extdata/refs/chr1.fa','/Users/runner/work/_temp/Library/Rhisat2/extdata/refs/chr2.fa','/Users/runner/work/_temp/Library/Rhisat2/extdata/refs/chr3.fa' '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpiccXQW/myindex'"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="aligning-reads-to-the-genome-index">Aligning reads to the genome index<a class="anchor" aria-label="anchor" href="#aligning-reads-to-the-genome-index"></a>
</h2>
<p>After creating the index, reads can be aligned using the <code>hisat2</code> wrapper function. Most commonly, the reads will be provided in <code>fastq</code> files (one file for single-end reads, two files for paired-end reads). The names of these files can be provided directly to the <code>hisat2</code> function, either as a vector (for single-end reads) or as a list of length 2 (for paired-end reads, each list element corresponds to one mate). You also need to provide the path to the index generated by <code>hisat2_build</code> (the output directory combined with the prefix), and the output file name where the alignments should be written.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/reads"</span>, package<span class="op">=</span><span class="st">"Rhisat2"</span><span class="op">)</span>, </span>
<span>           pattern<span class="op">=</span><span class="st">"\\.fastq$"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "reads1.fastq" "reads2.fastq"</span></span>
<span><span class="va">reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/reads"</span>, package<span class="op">=</span><span class="st">"Rhisat2"</span><span class="op">)</span>,</span>
<span>                    pattern<span class="op">=</span><span class="st">"\\.fastq$"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/hisat2.html">hisat2</a></span><span class="op">(</span>sequences<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">reads</span><span class="op">)</span>, index<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">td</span>, <span class="st">"myindex"</span><span class="op">)</span>, </span>
<span>       type<span class="op">=</span><span class="st">"paired"</span>, outfile<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">td</span>, <span class="st">"out.sam"</span><span class="op">)</span>, </span>
<span>       force<span class="op">=</span><span class="cn">TRUE</span>, strict<span class="op">=</span><span class="cn">TRUE</span>, execute<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>As for <code><a href="../reference/hisat2_build.html">hisat2_build()</a></code>, any additional arguments to the <code>hisat2</code> binary can be provided to the <code><a href="../reference/hisat2.html">hisat2()</a></code> function (see <code><a href="../reference/hisat2_usage.html">hisat2_usage()</a></code> or <a href="https://ccb.jhu.edu/software/hisat2/manual.shtml#running-hisat2" class="external-link">https://ccb.jhu.edu/software/hisat2/manual.shtml#running-hisat2</a> for an overview of the available options).</p>
<p>With HISAT2, you can provide a file with known splice sites at the alignment step, which can help in finding the correct alignments across known splice junctions. A text file in the required format can be generated using the <code><a href="../reference/extract_splice_sites.html">extract_splice_sites()</a></code> function, starting from an annotation file in gtf or gff3 format, or from a <code>GRanges</code> or <code>TxDb</code> object.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spsfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/refs/genes.gtf"</span>, package<span class="op">=</span><span class="st">"Rhisat2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/extract_splice_sites.html">extract_splice_sites</a></span><span class="op">(</span>features<span class="op">=</span><span class="va">gtf</span>, outfile<span class="op">=</span><span class="va">spsfile</span><span class="op">)</span></span>
<span><span class="co">#&gt; Import genomic features from the file as a GRanges object ... OK</span></span>
<span><span class="co">#&gt; Prepare the 'metadata' data frame ... OK</span></span>
<span><span class="co">#&gt; Make the TxDb object ... OK</span></span>
<span><span class="fu"><a href="../reference/hisat2.html">hisat2</a></span><span class="op">(</span>sequences<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">reads</span><span class="op">)</span>, index<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">td</span>, <span class="st">"myindex"</span><span class="op">)</span>,</span>
<span>       type<span class="op">=</span><span class="st">"paired"</span>, outfile<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">td</span>, <span class="st">"out_sps.sam"</span><span class="op">)</span>,</span>
<span>       `known-splicesite-infile`<span class="op">=</span><span class="va">spsfile</span>, </span>
<span>       force<span class="op">=</span><span class="cn">TRUE</span>, strict<span class="op">=</span><span class="cn">TRUE</span>, execute<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="miscellaneous-helper-functions">Miscellaneous helper functions<a class="anchor" aria-label="anchor" href="#miscellaneous-helper-functions"></a>
</h2>
<p>To get the version of <code>HISAT2</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hisat2_version.html">hisat2_version</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/Users/runner/work/_temp/Library/Rhisat2/hisat2 version 2.2.1"          </span></span>
<span><span class="co">#&gt; [2] "64-bit"                                                                 </span></span>
<span><span class="co">#&gt; [3] "Built on Mac-1666341372432.local"                                       </span></span>
<span><span class="co">#&gt; [4] "Fri Oct 21 08:50:38 UTC 2022"                                           </span></span>
<span><span class="co">#&gt; [5] "Compiler: gcc version 8.2.0 (GCC) "                                     </span></span>
<span><span class="co">#&gt; [6] "Options: -O3 -m64 -msse2 -funroll-loops -g3 -std=c++11"                 </span></span>
<span><span class="co">#&gt; [7] "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}"</span></span></code></pre></div>
<p>To see the usage of <code><a href="../reference/hisat2_build.html">hisat2_build()</a></code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hisat2_build_usage.html">hisat2_build_usage</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "HISAT2 version 2.2.1 by Daehwan Kim (infphilo@gmail.com, http://www.ccb.jhu.edu/people/infphilo)"</span></span>
<span><span class="co">#&gt;  [2] "Usage: hisat2-build [options]* &lt;reference_in&gt; &lt;ht2_index_base&gt;"                                  </span></span>
<span><span class="co">#&gt;  [3] "    reference_in            comma-separated list of files with ref sequences"                    </span></span>
<span><span class="co">#&gt;  [4] "    hisat2_index_base       write ht2 data to files with this dir/basename"                      </span></span>
<span><span class="co">#&gt;  [5] "Options:"                                                                                        </span></span>
<span><span class="co">#&gt;  [6] "    -c                      reference sequences given on cmd line (as"                           </span></span>
<span><span class="co">#&gt;  [7] "                            &lt;reference_in&gt;)"                                                     </span></span>
<span><span class="co">#&gt;  [8] "    -a/--noauto             disable automatic -p/--bmax/--dcv memory-fitting"                    </span></span>
<span><span class="co">#&gt;  [9] "    -p &lt;int&gt;                number of threads"                                                   </span></span>
<span><span class="co">#&gt; [10] "    --bmax &lt;int&gt;            max bucket sz for blockwise suffix-array builder"                    </span></span>
<span><span class="co">#&gt; [11] "    --bmaxdivn &lt;int&gt;        max bucket sz as divisor of ref len (default: 4)"                    </span></span>
<span><span class="co">#&gt; [12] "    --dcv &lt;int&gt;             diff-cover period for blockwise (default: 1024)"                     </span></span>
<span><span class="co">#&gt; [13] "    --nodc                  disable diff-cover (algorithm becomes quadratic)"                    </span></span>
<span><span class="co">#&gt; [14] "    -r/--noref              don't build .3/.4.ht2 (packed reference) portion"                    </span></span>
<span><span class="co">#&gt; [15] "    -3/--justref            just build .3/.4.ht2 (packed reference) portion"                     </span></span>
<span><span class="co">#&gt; [16] "    -o/--offrate &lt;int&gt;      SA is sampled every 2^offRate BWT chars (default: 5)"                </span></span>
<span><span class="co">#&gt; [17] "    -t/--ftabchars &lt;int&gt;    # of chars consumed in initial lookup (default: 10)"                 </span></span>
<span><span class="co">#&gt; [18] "    --localoffrate &lt;int&gt;    SA (local) is sampled every 2^offRate BWT chars (default: 3)"        </span></span>
<span><span class="co">#&gt; [19] "    --localftabchars &lt;int&gt;  # of chars consumed in initial lookup in a local index (default: 6)" </span></span>
<span><span class="co">#&gt; [20] "    --snp &lt;path&gt;            SNP file name"                                                       </span></span>
<span><span class="co">#&gt; [21] "    --haplotype &lt;path&gt;      haplotype file name"                                                 </span></span>
<span><span class="co">#&gt; [22] "    --ss &lt;path&gt;             Splice site file name"                                               </span></span>
<span><span class="co">#&gt; [23] "    --exon &lt;path&gt;           Exon file name"                                                      </span></span>
<span><span class="co">#&gt; [24] "    --repeat-ref &lt;path&gt;     Repeat reference file name"                                          </span></span>
<span><span class="co">#&gt; [25] "    --repeat-info &lt;path&gt;    Repeat information file name"                                        </span></span>
<span><span class="co">#&gt; [26] "    --repeat-snp &lt;path&gt;     Repeat snp file name"                                                </span></span>
<span><span class="co">#&gt; [27] "    --repeat-haplotype &lt;path&gt;   Repeat haplotype file name"                                      </span></span>
<span><span class="co">#&gt; [28] "    --seed &lt;int&gt;            seed for random number generator"                                    </span></span>
<span><span class="co">#&gt; [29] "    -q/--quiet              disable verbose output (for debugging)"                              </span></span>
<span><span class="co">#&gt; [30] "    -h/--help               print detailed description of tool and its options"                  </span></span>
<span><span class="co">#&gt; [31] "    --usage                 print this usage message"                                            </span></span>
<span><span class="co">#&gt; [32] "    --version               print version information and quit"</span></span></code></pre></div>
<p>And similarly to see the usage of <code><a href="../reference/hisat2.html">hisat2()</a></code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hisat2_usage.html">hisat2_usage</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [1] "HISAT2 version 2.2.1 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)"                                                                                                         </span></span>
<span><span class="co">#&gt;   [2] "Usage: "                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt;   [3] "  hisat2-align [options]* -x &lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]"                                                                                                                      </span></span>
<span><span class="co">#&gt;   [4] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;   [5] "  &lt;ht2-idx&gt;  Index filename prefix (minus trailing .X.ht2)."                                                                                                                                       </span></span>
<span><span class="co">#&gt;   [6] "  &lt;m1&gt;       Files with #1 mates, paired with files in &lt;m2&gt;."                                                                                                                                      </span></span>
<span><span class="co">#&gt;   [7] "  &lt;m2&gt;       Files with #2 mates, paired with files in &lt;m1&gt;."                                                                                                                                      </span></span>
<span><span class="co">#&gt;   [8] "  &lt;r&gt;        Files with unpaired reads."                                                                                                                                                           </span></span>
<span><span class="co">#&gt;   [9] "  &lt;sam&gt;      File for SAM output (default: stdout)"                                                                                                                                                </span></span>
<span><span class="co">#&gt;  [10] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [11] "  &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (no whitespace) and can be"                                                                                                                         </span></span>
<span><span class="co">#&gt;  [12] "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'."                                                                                                                                 </span></span>
<span><span class="co">#&gt;  [13] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [14] "Options (defaults in parentheses):"                                                                                                                                                                </span></span>
<span><span class="co">#&gt;  [15] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [16] " Input:"                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt;  [17] "  -q                 query input files are FASTQ .fq/.fastq (default)"                                                                                                                             </span></span>
<span><span class="co">#&gt;  [18] "  --qseq             query input files are in Illumina's qseq format"                                                                                                                              </span></span>
<span><span class="co">#&gt;  [19] "  -f                 query input files are (multi-)FASTA .fa/.mfa"                                                                                                                                 </span></span>
<span><span class="co">#&gt;  [20] "  -r                 query input files are raw one-sequence-per-line"                                                                                                                              </span></span>
<span><span class="co">#&gt;  [21] "  -c                 &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; are sequences themselves, not files"                                                                                                                          </span></span>
<span><span class="co">#&gt;  [22] "  -s/--skip &lt;int&gt;    skip the first &lt;int&gt; reads/pairs in the input (none)"                                                                                                                         </span></span>
<span><span class="co">#&gt;  [23] "  -u/--upto &lt;int&gt;    stop after first &lt;int&gt; reads/pairs (no limit)"                                                                                                                                </span></span>
<span><span class="co">#&gt;  [24] "  -5/--trim5 &lt;int&gt;   trim &lt;int&gt; bases from 5'/left end of reads (0)"                                                                                                                               </span></span>
<span><span class="co">#&gt;  [25] "  -3/--trim3 &lt;int&gt;   trim &lt;int&gt; bases from 3'/right end of reads (0)"                                                                                                                              </span></span>
<span><span class="co">#&gt;  [26] "  --phred33          qualities are Phred+33 (default)"                                                                                                                                             </span></span>
<span><span class="co">#&gt;  [27] "  --phred64          qualities are Phred+64"                                                                                                                                                       </span></span>
<span><span class="co">#&gt;  [28] "  --int-quals        qualities encoded as space-delimited integers"                                                                                                                                </span></span>
<span><span class="co">#&gt;  [29] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [30] " Presets:                 Same as:"                                                                                                                                                                </span></span>
<span><span class="co">#&gt;  [31] "   --fast                 --no-repeat-index"                                                                                                                                                       </span></span>
<span><span class="co">#&gt;  [32] "   --sensitive            --bowtie2-dp 1 -k 30 --score-min L,0,-0.5"                                                                                                                               </span></span>
<span><span class="co">#&gt;  [33] "   --very-sensitive       --bowtie2-dp 2 -k 50 --score-min L,0,-1"                                                                                                                                 </span></span>
<span><span class="co">#&gt;  [34] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [35] " Alignment:"                                                                                                                                                                                       </span></span>
<span><span class="co">#&gt;  [36] "  --bowtie2-dp &lt;int&gt; use Bowtie2's dynamic programming alignment algorithm (0) - 0: no dynamic programming, 1: conditional dynamic programming, and 2: unconditional dynamic programming (slowest)"</span></span>
<span><span class="co">#&gt;  [37] "  --n-ceil &lt;func&gt;    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)"                                                                                                                      </span></span>
<span><span class="co">#&gt;  [38] "  --ignore-quals     treat all quality values as 30 on Phred scale (off)"                                                                                                                          </span></span>
<span><span class="co">#&gt;  [39] "  --nofw             do not align forward (original) version of read (off)"                                                                                                                        </span></span>
<span><span class="co">#&gt;  [40] "  --norc             do not align reverse-complement version of read (off)"                                                                                                                        </span></span>
<span><span class="co">#&gt;  [41] "  --no-repeat-index  do not use repeat index"                                                                                                                                                      </span></span>
<span><span class="co">#&gt;  [42] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [43] " Spliced Alignment:"                                                                                                                                                                               </span></span>
<span><span class="co">#&gt;  [44] "  --pen-cansplice &lt;int&gt;              penalty for a canonical splice site (0)"                                                                                                                      </span></span>
<span><span class="co">#&gt;  [45] "  --pen-noncansplice &lt;int&gt;           penalty for a non-canonical splice site (12)"                                                                                                                 </span></span>
<span><span class="co">#&gt;  [46] "  --pen-canintronlen &lt;func&gt;          penalty for long introns (G,-8,1) with canonical splice sites"                                                                                                </span></span>
<span><span class="co">#&gt;  [47] "  --pen-noncanintronlen &lt;func&gt;       penalty for long introns (G,-8,1) with noncanonical splice sites"                                                                                             </span></span>
<span><span class="co">#&gt;  [48] "  --min-intronlen &lt;int&gt;              minimum intron length (20)"                                                                                                                                   </span></span>
<span><span class="co">#&gt;  [49] "  --max-intronlen &lt;int&gt;              maximum intron length (500000)"                                                                                                                               </span></span>
<span><span class="co">#&gt;  [50] "  --known-splicesite-infile &lt;path&gt;   provide a list of known splice sites"                                                                                                                         </span></span>
<span><span class="co">#&gt;  [51] "  --novel-splicesite-outfile &lt;path&gt;  report a list of splice sites"                                                                                                                                </span></span>
<span><span class="co">#&gt;  [52] "  --novel-splicesite-infile &lt;path&gt;   provide a list of novel splice sites"                                                                                                                         </span></span>
<span><span class="co">#&gt;  [53] "  --no-temp-splicesite               disable the use of splice sites found"                                                                                                                        </span></span>
<span><span class="co">#&gt;  [54] "  --no-spliced-alignment             disable spliced alignment"                                                                                                                                    </span></span>
<span><span class="co">#&gt;  [55] "  --rna-strandness &lt;string&gt;          specify strand-specific information (unstranded)"                                                                                                             </span></span>
<span><span class="co">#&gt;  [56] "  --tmo                              reports only those alignments within known transcriptome"                                                                                                     </span></span>
<span><span class="co">#&gt;  [57] "  --dta                              reports alignments tailored for transcript assemblers"                                                                                                        </span></span>
<span><span class="co">#&gt;  [58] "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks"                                                                                                       </span></span>
<span><span class="co">#&gt;  [59] "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)"                                                                                           </span></span>
<span><span class="co">#&gt;  [60] "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads"                                                                                                        </span></span>
<span><span class="co">#&gt;  [61] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [62] " Scoring:"                                                                                                                                                                                         </span></span>
<span><span class="co">#&gt;  [63] "  --mp &lt;int&gt;,&lt;int&gt;   max and min penalties for mismatch; lower qual = lower penalty &lt;6,2&gt;"                                                                                                         </span></span>
<span><span class="co">#&gt;  [64] "  --sp &lt;int&gt;,&lt;int&gt;   max and min penalties for soft-clipping; lower qual = lower penalty &lt;2,1&gt;"                                                                                                    </span></span>
<span><span class="co">#&gt;  [65] "  --no-softclip      no soft-clipping"                                                                                                                                                             </span></span>
<span><span class="co">#&gt;  [66] "  --np &lt;int&gt;         penalty for non-A/C/G/Ts in read/ref (1)"                                                                                                                                     </span></span>
<span><span class="co">#&gt;  [67] "  --rdg &lt;int&gt;,&lt;int&gt;  read gap open, extend penalties (5,3)"                                                                                                                                        </span></span>
<span><span class="co">#&gt;  [68] "  --rfg &lt;int&gt;,&lt;int&gt;  reference gap open, extend penalties (5,3)"                                                                                                                                   </span></span>
<span><span class="co">#&gt;  [69] "  --score-min &lt;func&gt; min acceptable alignment score w/r/t read length"                                                                                                                             </span></span>
<span><span class="co">#&gt;  [70] "                     (L,0.0,-0.2)"                                                                                                                                                                 </span></span>
<span><span class="co">#&gt;  [71] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [72] " Reporting:"                                                                                                                                                                                       </span></span>
<span><span class="co">#&gt;  [73] "  -k &lt;int&gt;           It searches for at most &lt;int&gt; distinct, primary alignments for each read. Primary alignments mean "                                                                           </span></span>
<span><span class="co">#&gt;  [74] "                     alignments whose alignment score is equal to or higher than any other alignments. The search terminates "                                                                     </span></span>
<span><span class="co">#&gt;  [75] "                     when it cannot find more distinct valid alignments, or when it finds &lt;int&gt;, whichever happens first. "                                                                        </span></span>
<span><span class="co">#&gt;  [76] "                     The alignment score for a paired-end alignment equals the sum of the alignment scores of "                                                                                    </span></span>
<span><span class="co">#&gt;  [77] "                     the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit "                                                                     </span></span>
<span><span class="co">#&gt;  [78] "                     (which equals 256) set in its FLAGS field. For reads that have more than &lt;int&gt; distinct, "                                                                                    </span></span>
<span><span class="co">#&gt;  [79] "                     valid alignments, hisat2 does not guarantee that the &lt;int&gt; alignments reported are the best possible "                                                                        </span></span>
<span><span class="co">#&gt;  [80] "                     in terms of alignment score. Default: 5 (linear index) or 10 (graph index)."                                                                                                  </span></span>
<span><span class="co">#&gt;  [81] "                     Note: HISAT2 is not designed with large values for -k in mind, and when aligning reads to long, "                                                                             </span></span>
<span><span class="co">#&gt;  [82] "                     repetitive genomes, large -k could make alignment much slower."                                                                                                               </span></span>
<span><span class="co">#&gt;  [83] "  --max-seeds &lt;int&gt;  HISAT2, like other aligners, uses seed-and-extend approaches. HISAT2 tries to extend seeds to "                                                                               </span></span>
<span><span class="co">#&gt;  [84] "                     full-length alignments. In HISAT2, --max-seeds is used to control the maximum number of seeds that "                                                                          </span></span>
<span><span class="co">#&gt;  [85] "                     will be extended. For DNA-read alignment (--no-spliced-alignment), HISAT2 extends up to these many seeds"                                                                     </span></span>
<span><span class="co">#&gt;  [86] "                     and skips the rest of the seeds. For RNA-read alignment, HISAT2 skips extending seeds and reports "                                                                           </span></span>
<span><span class="co">#&gt;  [87] "                     no alignments if the number of seeds is larger than the number specified with the option, "                                                                                   </span></span>
<span><span class="co">#&gt;  [88] "                     to be compatible with previous versions of HISAT2. Large values for --max-seeds may improve alignment "                                                                       </span></span>
<span><span class="co">#&gt;  [89] "                     sensitivity, but HISAT2 is not designed with large values for --max-seeds in mind, and when aligning "                                                                        </span></span>
<span><span class="co">#&gt;  [90] "                     reads to long, repetitive genomes, large --max-seeds could make alignment much slower. "                                                                                      </span></span>
<span><span class="co">#&gt;  [91] "                     The default value is the maximum of 5 and the value that comes with -k times 2."                                                                                              </span></span>
<span><span class="co">#&gt;  [92] "  -a/--all           HISAT2 reports all alignments it can find. Using the option is equivalent to using both --max-seeds "                                                                         </span></span>
<span><span class="co">#&gt;  [93] "                     and -k with the maximum value that a 64-bit signed integer can represent (9,223,372,036,854,775,807)."                                                                        </span></span>
<span><span class="co">#&gt;  [94] "  --repeat           report alignments to repeat sequences directly"                                                                                                                               </span></span>
<span><span class="co">#&gt;  [95] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt;  [96] " Paired-end:"                                                                                                                                                                                      </span></span>
<span><span class="co">#&gt;  [97] "  -I/--minins &lt;int&gt;  minimum fragment length (0), only valid with --no-spliced-alignment"                                                                                                          </span></span>
<span><span class="co">#&gt;  [98] "  -X/--maxins &lt;int&gt;  maximum fragment length (500), only valid with --no-spliced-alignment"                                                                                                        </span></span>
<span><span class="co">#&gt;  [99] "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)"                                                                                                                              </span></span>
<span><span class="co">#&gt; [100] "  --no-mixed         suppress unpaired alignments for paired reads"                                                                                                                                </span></span>
<span><span class="co">#&gt; [101] "  --no-discordant    suppress discordant alignments for paired reads"                                                                                                                              </span></span>
<span><span class="co">#&gt; [102] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt; [103] " Output:"                                                                                                                                                                                          </span></span>
<span><span class="co">#&gt; [104] "  -t/--time          print wall-clock time taken by search phases"                                                                                                                                 </span></span>
<span><span class="co">#&gt; [105] "  --summary-file &lt;path&gt; print alignment summary to this file."                                                                                                                                     </span></span>
<span><span class="co">#&gt; [106] "  --new-summary         print alignment summary in a new style, which is more machine-friendly."                                                                                                   </span></span>
<span><span class="co">#&gt; [107] "  --quiet               print nothing to stderr except serious errors"                                                                                                                             </span></span>
<span><span class="co">#&gt; [108] "  --met-file &lt;path&gt;     send metrics to file at &lt;path&gt; (off)"                                                                                                                                      </span></span>
<span><span class="co">#&gt; [109] "  --met-stderr          send metrics to stderr (off)"                                                                                                                                              </span></span>
<span><span class="co">#&gt; [110] "  --met &lt;int&gt;           report internal counters &amp; metrics every &lt;int&gt; secs (1)"                                                                                                                   </span></span>
<span><span class="co">#&gt; [111] "  --no-head             suppress header lines, i.e. lines starting with @"                                                                                                                         </span></span>
<span><span class="co">#&gt; [112] "  --no-sq               suppress @SQ header lines"                                                                                                                                                 </span></span>
<span><span class="co">#&gt; [113] "  --rg-id &lt;text&gt;        set read group id, reflected in @RG line and RG:Z: opt field"                                                                                                              </span></span>
<span><span class="co">#&gt; [114] "  --rg &lt;text&gt;           add &lt;text&gt; (\"lab:value\") to @RG line of SAM header."                                                                                                                     </span></span>
<span><span class="co">#&gt; [115] "                        Note: @RG line only printed when --rg-id is set."                                                                                                                          </span></span>
<span><span class="co">#&gt; [116] "  --omit-sec-seq        put '*' in SEQ and QUAL fields for secondary alignments."                                                                                                                  </span></span>
<span><span class="co">#&gt; [117] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt; [118] " Performance:"                                                                                                                                                                                     </span></span>
<span><span class="co">#&gt; [119] "  -o/--offrate &lt;int&gt; override offrate of index; must be &gt;= index's offrate"                                                                                                                        </span></span>
<span><span class="co">#&gt; [120] "  -p/--threads &lt;int&gt; number of alignment threads to launch (1)"                                                                                                                                    </span></span>
<span><span class="co">#&gt; [121] "  --reorder          force SAM output order to match order of input reads"                                                                                                                         </span></span>
<span><span class="co">#&gt; [122] "  --mm               use memory-mapped I/O for index; many 'hisat2's can share"                                                                                                                    </span></span>
<span><span class="co">#&gt; [123] ""                                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt; [124] " Other:"                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt; [125] "  --qc-filter        filter out reads that are bad according to QSEQ filter"                                                                                                                       </span></span>
<span><span class="co">#&gt; [126] "  --seed &lt;int&gt;       seed for random number generator (0)"                                                                                                                                         </span></span>
<span><span class="co">#&gt; [127] "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes"                                                                                                                </span></span>
<span><span class="co">#&gt; [128] "  --remove-chrname   remove 'chr' from reference names in alignment"                                                                                                                               </span></span>
<span><span class="co">#&gt; [129] "  --add-chrname      add 'chr' to reference names in alignment "                                                                                                                                   </span></span>
<span><span class="co">#&gt; [130] "  --version          print version information and quit"                                                                                                                                           </span></span>
<span><span class="co">#&gt; [131] "  -h/--help          print this usage message"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] Rhisat2_1.13.1   BiocStyle_2.25.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-7                matrixStats_0.62.0          fs_1.5.2                   </span></span>
<span><span class="co">#&gt;   [4] bit64_4.0.5                 filelock_1.0.2              progress_1.2.2             </span></span>
<span><span class="co">#&gt;   [7] httr_1.4.4                  rprojroot_2.0.3             GenomeInfoDb_1.33.13       </span></span>
<span><span class="co">#&gt;  [10] tools_4.2.1                 bslib_0.4.0                 utf8_1.2.2                 </span></span>
<span><span class="co">#&gt;  [13] R6_2.5.1                    DBI_1.1.3                   BiocGenerics_0.43.4        </span></span>
<span><span class="co">#&gt;  [16] tidyselect_1.2.0            prettyunits_1.1.1           bit_4.0.4                  </span></span>
<span><span class="co">#&gt;  [19] curl_4.3.3                  compiler_4.2.1              SGSeq_1.31.0               </span></span>
<span><span class="co">#&gt;  [22] textshaping_0.3.6           cli_3.4.1                   Biobase_2.57.1             </span></span>
<span><span class="co">#&gt;  [25] xml2_1.3.3                  desc_1.4.2                  DelayedArray_0.23.2        </span></span>
<span><span class="co">#&gt;  [28] rtracklayer_1.57.0          bookdown_0.29               sass_0.4.2                 </span></span>
<span><span class="co">#&gt;  [31] rappdirs_0.3.3              pkgdown_2.0.6.9000          systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt;  [34] stringr_1.4.1               digest_0.6.30               Rsamtools_2.13.4           </span></span>
<span><span class="co">#&gt;  [37] rmarkdown_2.17              XVector_0.37.1              pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [40] htmltools_0.5.3             MatrixGenerics_1.9.1        dbplyr_2.2.1               </span></span>
<span><span class="co">#&gt;  [43] fastmap_1.1.0               rlang_1.0.6                 RSQLite_2.2.18             </span></span>
<span><span class="co">#&gt;  [46] jquerylib_0.1.4             BiocIO_1.7.1                generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [49] jsonlite_1.8.2              BiocParallel_1.31.14        dplyr_1.0.10               </span></span>
<span><span class="co">#&gt;  [52] RCurl_1.98-1.9              magrittr_2.0.3              GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">#&gt;  [55] Matrix_1.5-1                Rcpp_1.0.9                  S4Vectors_0.35.4           </span></span>
<span><span class="co">#&gt;  [58] fansi_1.0.3                 lifecycle_1.0.3             stringi_1.7.8              </span></span>
<span><span class="co">#&gt;  [61] yaml_2.3.6                  SummarizedExperiment_1.27.3 zlibbioc_1.43.0            </span></span>
<span><span class="co">#&gt;  [64] BiocFileCache_2.5.2         grid_4.2.1                  blob_1.2.3                 </span></span>
<span><span class="co">#&gt;  [67] parallel_4.2.1              crayon_1.5.2                lattice_0.20-45            </span></span>
<span><span class="co">#&gt;  [70] Biostrings_2.65.6           GenomicFeatures_1.49.7      hms_1.1.2                  </span></span>
<span><span class="co">#&gt;  [73] KEGGREST_1.37.3             knitr_1.40                  pillar_1.8.1               </span></span>
<span><span class="co">#&gt;  [76] igraph_1.3.5                RUnit_0.4.32                GenomicRanges_1.49.1       </span></span>
<span><span class="co">#&gt;  [79] rjson_0.2.21                codetools_0.2-18            biomaRt_2.53.3             </span></span>
<span><span class="co">#&gt;  [82] stats4_4.2.1                XML_3.99-0.11               glue_1.6.2                 </span></span>
<span><span class="co">#&gt;  [85] evaluate_0.17               BiocManager_1.30.18         png_0.1-7                  </span></span>
<span><span class="co">#&gt;  [88] vctrs_0.4.2                 purrr_0.3.5                 assertthat_0.2.1           </span></span>
<span><span class="co">#&gt;  [91] cachem_1.0.6                xfun_0.34                   restfulr_0.0.15            </span></span>
<span><span class="co">#&gt;  [94] ragg_1.2.3                  tibble_3.1.8                GenomicAlignments_1.33.1   </span></span>
<span><span class="co">#&gt;  [97] AnnotationDbi_1.59.1        memoise_2.0.1               IRanges_2.31.2             </span></span>
<span><span class="co">#&gt; [100] ellipsis_0.3.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Kim2015-mu" class="csl-entry">
Kim, Daehwan, Ben Langmead, and Steven L Salzberg. 2015. <span>“<span>HISAT</span>: A Fast Spliced Aligner with Low Memory Requirements.”</span> <em>Nat. Methods</em> 12: 357.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Charlotte Soneson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
